### ISSUE TITLE
"My Datasets" should include datasets in which the user is a creator

### ISSUE THREAD

**Issue by laurenwalker** [NCEAS/metacatui#1160]
Right now if you click on "My Datasets" or go to your profile, a search is conducted that searches for datasets for which you are indexed as the submitter, writePermission, changePermission, or rightsHolder. But should we also list datasets for which you are listed as a creator, but you don't have write access to? During the DataONE data training this week, Matt got some helpful feedback that people were expecting this behavior.

This will make the My Datasets list more inclusive, but it also means you may not be able to edit every dataset in the list. We'll want to make sure our UI clearly communicates that.

**Comment by amoeba** [comment]
> But should we also list datasets for which you are listed as a creator, but you don't have write access to?

I'd say 👍, though I think the reason we've never done this is at least in part because DataONE rarely has a person identifier to go with the Creator (<1%) so the subject of the person that's logged in with is unlikely to be matched within the EML record. Plus, we don't index user person identifiers so this query is currently expensive.

If we start indexing EML party `userId`'s for `creator`, this feature would work really well on the KNB and ADC since login and the new editor use ORCIDs. And if we did this, maybe it'd be good to let the user filter by these different mappings (submitter, rightsHolder, etc) with clear explanations like you suggest.

**Comment by laurenwalker** [comment]
Agreed on the indexing, Bryce. I'll create a Metacat ticket for that.

It would also be great to have a feature to "request access", which sends a notification to the dataset owner that someone wants write access. We don't have a notification structure in place right now, but a fallback could be to display the owner's contact info (taken from the metadata) and some basic instructions.

**Comment by mbjones** [comment]
I agree we should also make it easy for creators to request access to datasets on which they are listed as creators.

We've also talked about enriching our cross-linking among people from the metadata and the account service so that we can better resolve, say, an account ID, an ORCID, and the creator's name and email.  It would be really useful if this information were indexed and accessible, so that if we find an email or ORCID in EML, it can get linked to an account.
